### PR TITLE
Fix string to objectId mapping error when using query method.

### PR DESCRIPTION
**Pull request by grwang91**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

Fixed objectId conversion issue.
Merged common codes into method.

Resolves: #4490

### CODE REVIEW

**Comment by christophstrobl** [comment]
@grwang91 thank you for the the PR!

**Comment by christophstrobl** [comment]
I think I'd like to see the changes contained in #4517 go in first and then bring in additional logic for handling the `id` conversion from this PR.

**Comment by grwang91** [comment]
> I think I'd like to see the changes contained in #4517 go in first and then bring in additional logic for handling the `id` conversion from this PR.

Do you mean applying this PR after merging #4518 to main branch?

File: spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/QueryMapper.java [hunk 1]
```diff
@@ -517,10 +492,44 @@ else if (isDocument(value)) {
...
+		if (valueDbo.containsKey("$in") || valueDbo.containsKey("$nin")) {
+			String inKey = valueDbo.containsKey("$in") ? "$in" : "$nin";
+			List<Object> ids = new ArrayList<>();
+			for (Object id : (Iterable<?>) valueDbo.get(inKey)) {
+				ids.add(convertId(id, getIdTypeForField(documentField)));
+			}
+			resultDbo.put(inKey, ids);
+		} else if (valueDbo.containsKey("$ne")) {
+			resultDbo.put("$ne", convertId(valueDbo.get("$ne"), getIdTypeForField(documentField)));
+		} else if (containsCompareOperator(valueDbo)) {
```
[the remark this answers is not in the document]
Comment (grwang91) [reply spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/QueryMapper.java:505]: Thanks for suggestion. I will apply it!

**Comment by mp911de** [comment]
Thank you for your contribution. That's merged and polished now.

**Pull request closed without merging**